### ISSUE TITLE
Spec for Memory object's constructor does not clamp maximum >= initial

### ISSUE THREAD

**Issue by lars-t-hansen** [WebAssembly/spec#876]
The descriptor passed to WebAssembly.Memory's constructor has two fields, initial and maximum, where maximum is optional.  If maximum is present then there needs to be a check that it is larger than initial, but there's no such check.  WebAssembly.Table's constructor does have such a check.

**Comment by lukewagner** [comment]
Good spot!  It seems like ideally this validation of the limits would happen using the core spec's [validation rule for memory types](https://webassembly.github.io/spec/core/valid/types.html#memory-types) and [table types](https://webassembly.github.io/spec/core/valid/types.html#table-types).  Maybe the simplest place for this might be in the core spec [`alloc_mem`](https://webassembly.github.io/spec/core/appendix/embedding.html#embed-alloc-mem) and [`alloc_table`](https://webassembly.github.io/spec/core/appendix/embedding.html#embed-alloc-table) functions, so that the wasm core spec is preserving its own validation invariants?  WDYT @rossberg?

**Comment by lukewagner** [comment]
(Oops, I hadn't reached #877 yet in my bugmail.)  Either approach works, I suppose.

**Comment by rossberg** [comment]
@lukewagner, yes it would make sense to add these as explicit pre/side conditions. I'll do that.
